src/Authenticator/Authenticator.php

```diff
@@ -37,7 +38,8 @@ public function checkPassword(Member $member, $password, ValidationResult &$resu
 
     public function getLogoutHandler($link)
     {
-        // No-op
+        // return the default handler
+        return LogoutHandler::create();
```

Comment:
```suggestion
        return LogoutHandler::create($link, $this);
```

Reply:
Wait, `SilverStripe\Security\MemberAuthenticator\LogoutHandler`
does [not have own constructor](https://github.com/silverstripe/silverstripe-framework/blob/4.11.15/src/Security/MemberAuthenticator/LogoutHandler.php)
inherits a constructor from `SilverStripe\Control\RequestHandler` which [does not accept any parameters](https://github.com/silverstripe/silverstripe-framework/blob/4.11.15/src/Control/RequestHandler.php#L121).
At leas not in `silverstripe/framework` 4.11.15

Reply:
Yep, it’s just for consistency with core - I assume the original intent was that if someone replaces the “handler” with their own service via Injector, they _could_ add those params

https://github.com/silverstripe/silverstripe-framework/blob/4e1b99b8c710bb8d88da91aecbf4d28d86146008/src/Security/MemberAuthenticator/MemberAuthenticator.php#L227-L261